src/functions.php

```diff
@@ -601,3 +661,17 @@ function waterfall(array $tasks): PromiseInterface
 
     return $deferred->promise();
 }
+
+/**
+ * @internal
+ */
+function fiberMap(): FiberMap
+{
+    static $wm = null;
```

Comment:
Using a static variable here is an *interesting* design choice. It's my understanding this will create a single object reference that will only be freed once the PHP engine shuts down. I wonder if it makes more sense to use static methods on the `FiberMap` instead of avoid this garbage reference and avoid any additional function call indirections here?

Reply:
Good point, that works just as well :+1: